### PR TITLE
Publish kdbush definition

### PR DESCRIPTION
**Pull request by DenisCarriere**
## [kdbush](https://github.com/mourner/kdbush)
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/kdbush/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Type definitions for kdbush 1.0
+// Project: https://github.com/mourner/kdbush
+// Definitions by: DenisCarriere <https://github.com/DenisCarriere>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type Points = number[][];
+type Get<T> = (p: T) => void;
```
Comment (andy-ms) [inline types/kdbush/index.d.ts]: Based on the documentation this looks like it should be `(point: T) => number`.

File: types/kdbush/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Type definitions for kdbush 1.0
+// Project: https://github.com/mourner/kdbush
+// Definitions by: DenisCarriere <https://github.com/DenisCarriere>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type Points = number[][];
```
Comment (andy-ms) [inline types/kdbush/index.d.ts]: `Array<[number, number]>`?

File: types/kdbush/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  coords: number[];
+  nodeSize: number;
+  points: T[];
+  range(minX: number, minY: number, maxX: number, maxY: number): number[];
+  within(x: number, y: number, r: number): number[];
+}
+
+interface KDBushStatic {
+  (points: Points): KDBush<Points>;
+  <T>(points: T[], getX: Get<T>, getY: Get<T>, nodeSize?: number, ArrayType?: any): KDBush<T>;
```
Comment (andy-ms) [inline types/kdbush/index.d.ts]: It looks like there are only a few possiblilities for `ArrayType` -- could it just be `typeof Int32Array | typeof Array`?

File: types/kdbush/index.d.ts (hunk 2 above)
Comment (DenisCarriere) [inline types/kdbush/index.d.ts]: True, however 95% of the time some one will be using [`@types/geojson`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/geojson) and the [Point types are defined as `number[]`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/geojson/index.d.ts#L20).

File: types/kdbush/index.d.ts (hunk 3 above)
Comment (DenisCarriere) [inline types/kdbush/index.d.ts]: 👍 Yes agreed, I didn't know how to define ArrayTypes (Thanks for your input on that one!)

File: types/kdbush/index.d.ts (hunk 2 above)
Comment (DenisCarriere) [inline types/kdbush/index.d.ts]: Also including a `z` value in your points is also valid and does not require having the `getX` & `getY` callbacks.

```javascript
// z coordinate (does not require Get callbacks)
const pointsZ = [[110, 60, 4000], [130, 40, 3000]];
kdbush(pointsZ);
```

**Comment by DenisCarriere** [comment]
@andy-ms Made some changes to the definition based on your comments. 👍 Thanks for your feedback.

File: types/kdbush/index.d.ts (hunk 1 above)
Comment (DenisCarriere) [inline types/kdbush/index.d.ts]: 👍 Agreed

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**